Applet/JCardSimProviderLib/src/com/android/javacard/seprovider/KMJCardSimulator.java

```diff
@@ -1653,12 +1653,12 @@ public KMKey createMasterKey(KMKey masterKey, short keySizeBits) {
     if (masterKey == null) {
```

Comment:
I think instead of removing the if - else isn't it better to add additional check that if the master key is null or if the master key length is zero then regenerate the key.